src/components/UserNicknamesPanel/UserNicknamesPanel.js

```diff
@@ -57,6 +57,19 @@ function UserNicknamesPanel () {
     return undefined
   }, [dispatch, nicknames, user])
 
+  const handleSetDisplayNickname = useCallback(async (event) => {
+    const nickname = nicknames.find((nick) => {
+      return nick.id === event.target.name
+    })
+
+    const response = await dispatch(setDisplayNickname(nickname.id, nickname.attributes.nick))
+
+    if (!isError(response)) {
+      // Refresh user profile to get updated nicknames
+      await dispatch(getUserProfile())
+    }
```

Comment:
Sorry I forgot about this, but because the way I handled loading states with confirmation buttons is weird, this function always needs to `return true`, or at least some truthy value. Otherwise we get an infinite spinner.